Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp

```diff
@@ -515,18 +514,36 @@ namespace Multiplayer
          || GetAgentType() == MultiplayerAgentType::DedicatedServer)
         {
             // We use a temporary userId over the clients address so we can maintain client lookups even in the event of wifi handoff
-            NetworkEntityHandle controlledEntity = SpawnDefaultPlayerPrefab(packet.GetTemporaryUserId());
-            EnableAutonomousControl(controlledEntity, connection->GetConnectionId());
+            IMultiplayerSpawner* spawner = AZ::Interface<IMultiplayerSpawner>::Get();
+            NetworkEntityHandle controlledEntity;
 
-            ServerToClientConnectionData* connectionData = reinterpret_cast<ServerToClientConnectionData*>(connection->GetUserData());
-            AZStd::unique_ptr<IReplicationWindow> window = AZStd::make_unique<ServerToClientReplicationWindow>(controlledEntity, connection);
-            connectionData->GetReplicationManager().SetReplicationWindow(AZStd::move(window));
-            connectionData->SetControlledEntity(controlledEntity);
+            // Check rejoin data first
+            const auto node = m_playerRejoinData.find(packet.GetTemporaryUserId());
+            if (node != m_playerRejoinData.end())
+            {
+                controlledEntity = m_networkEntityManager.GetNetworkEntityTracker()->Get(node->second);
+            }
+            else if (spawner)
+            {
+                // Route to spawner implementation
+                AZStd::pair<PrefabEntityId, AZ::Transform> spawnParams = spawner->OnPlayerJoin(packet.GetTemporaryUserId());
+                controlledEntity = TrySpawnPlayerPrefab(spawnParams.first, spawnParams.second);
+            }
```

Comment:
Pop out some kind of a visible error if you don't have a spawner. While I don't think we should support a cvar fallback path, we should make misconfiguration easy to detect and diagnose.